runtime/lua/vim/lsp.lua

```diff
@@ -871,7 +873,7 @@ function lsp.start(config, opts)
   if not config.name and type(config.cmd) == 'table' then
     config.name = config.cmd[1] and vim.fs.basename(config.cmd[1]) or nil
   end
-  local bufnr = api.nvim_get_current_buf()
+  local bufnr = opts.bufnr or api.nvim_get_current_buf()
```

Comment:
Is this the "proper" way to support a `bufnr=0` or does it need to be something like:

```lua
local bufnr = (opts.bufnr ~= nil and opts.bufnr ~= 0) and opts.bufnr or api.nvim_get_current_buf()
```

Reply:
As far as I can tell, the more common pattern is to do this over multiple lines:

```lua
local bufnr = opts.bufnr
if bufnr ==nil or bufnr == 0 then
  bufnr = api.nvim_get_current_buf()
end
```

You can see that pattern in multiple places in the neovim source code.

Reply:
Cool, updated to use that pattern for now. Will see where that goes.